fbpic/particles/injection/continuous_injection.py

```diff
@@ -168,8 +167,17 @@ def generate_particles( self, time ):
         # Create a temporary density function that takes into
         # account the fact that the plasma has moved
         if self.dens_func is not None:
-            def dens_func( z, r ):
-                return( self.dens_func( z-self.v_end_plasma*time, r ) )
+            if self.dens_func.__doc__ == 'xyz':
+                def dens_func( x, y, z ):
+                    return( self.dens_func(x, y, z-self.v_end_plasma*time ) )
+            elif self.dens_func.__doc__ == 'zrt':
+                def dens_func( z, r, th ):
+                    return( self.dens_func(z-self.v_end_plasma*time, r, th ) )
+            else:
+                def dens_func( z, r ):
+                    return( self.dens_func( z-self.v_end_plasma*time, r ) )
```

Comment:
Cool idea!
However, instead of using `__doc__`, could we check the names in `self.dens_func.__code__.co_varnames`?

Reply:
well, that's  indeed a more clear way! 
The drawback is that it won't do when function is defined as a class with the `__call__` method (as it's done in `density_profiles.py` and laser profiles). In this case `dens_func` won't have `__code__` attribute..

On one hand, I can adapt the behaviour of `density_profiles.py`, but I though that making `dens_func` as a class (like laser profile) could be an option for future development..

Reply:
@RemiLehe whenever you have time, let me know which way is preferable -- for me personally, setting `dens_func.__doc__='xzy'` in the input is a more explicit (=deliberate) way of doing things, but if you prefer to have it as an out-of-box feature, I'll adjust the PR